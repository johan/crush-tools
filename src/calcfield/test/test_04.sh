test_number=04
description="placement of calculated field"

infile="$test_dir/test.txt"

subtests=('-p -i 1' '-p -r 1' '-B Text2' '-A Metric-1')
subtest_desc=('insert first' 'replace first' 'before label' 'after label')

for i in `seq 0 $(( ${#subtests[*]} - 1 ))`; do
  expected="$test_dir/test_$test_number.$i.expected"
  outfile="$test_dir/test_$test_number.$i.actual"
  $bin ${subtests[$i]} -d ';' -e '[3] / [2]' -b '0.0' $infile > "$outfile"
  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $i "$description (${subtest_desc[$i]})" FAIL
  else
    test_status $test_number $i "$description (${subtest_desc[$i]})" PASS
    rm "$outfile"
  fi
done
