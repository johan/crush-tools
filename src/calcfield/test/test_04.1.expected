Calculated-Field;Metric-0;Metric-1;Text2
3;1;3;world
0.142857142857143;7;1;wolff
0;400;0;test
